front/src/applications/operationalStudies/components/ManageTrainSchedule/SpeedLimitByTagSelector.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { ComponentType, useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useTranslation } from 'react-i18next';
+import { updateSpeedLimitByTag } from 'reducers/osrdconf';
+import { setFailure } from 'reducers/main';
+import icon from 'assets/pictures/components/speedometer.svg';
+import SelectImprovedSNCF from 'common/BootstrapSNCF/SelectImprovedSNCF';
+import { getInfraID, getSpeedLimitByTag } from 'reducers/osrdconf/selectors';
+import { Dispatch } from 'redux';
+import { noop } from 'lodash';
+import { osrdMiddlewareApi } from 'common/api/osrdMiddlewareApi';
+import { FetchBaseQueryError } from '@reduxjs/toolkit/query/react';
+
+type SpeedLimitByTagSelectorProps = {
+  condensed?: boolean;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  speedLimitByTag?: string;
+  dispatch?: Dispatch;
+  speedLimitsByTagsFromApi?: string[];
+};
+
+function withOSRDData<T>(Component: ComponentType<T>) {
+  return (hocProps: T) => {
+    const { t } = useTranslation(['operationalStudies/manageTrainSchedule']);
+    const dispatch = useDispatch();
+    const infraID = useSelector(getInfraID);
+    const speedLimitByTag = useSelector(getSpeedLimitByTag);
+    const { data, error } = osrdMiddlewareApi.useGetInfraByIdSpeedLimitTagsQuery(
+      {
+        id: infraID as number,
+      },
+      { skip: !infraID }
+    );
+
+    useEffect(() => {
+      // Update the document title using the browser API
+      if (error) {
+        dispatch(
+          setFailure({
+            name: t('errorMessages.unableToRetrieveTags'),
+            message: `${(error as FetchBaseQueryError).status} : ${JSON.stringify(
+              (error as FetchBaseQueryError).data
+            )}`,
+          })
+        );
+      }
+    }, [error]);
+
+    return (
+      <Component
+        {...(hocProps as T)}
+        dispatch={dispatch}
+        speedLimitsByTagsFromApi={data}
+        speedLimitByTag={speedLimitByTag}
+      />
+    );
+  };
+}
+
+export function SpeedLimitByTagSelector({
```

Comment:
```suggestion
export function IsolatedSpeedLimitByTagSelector({
```
Can you rename this function IsolatedSpeedLimitByTagSelector ? Otherwise, you have a warning when you import SpeedLimitByTagSelector (default import) in ManageTrainSchedule for instance

Reply:
Done